ci_framework/plugins/action/oc.py

```diff
@@ -0,0 +1,113 @@
+# Copyright Red Hat, Inc.
+# Apache License Version 2.0 (see LICENSE)
+
+import os
+from pathlib import Path
+from ansible.errors import AnsibleError
+from ansible.module_utils.basic import get_bin_path
+from ansible.module_utils.six import ensure_text
+from ansible.plugins.action import ActionBase
+
+DOCUMENTATION = r'''
+---
+action: oc
+
+short_description: Execute OpensShift Client commands - oc -.
+
+description:
+- This module searches the oc binary, load the kubeconfig and execute the oc provided commands.
+'''
+
+EXAMPLES = r'''
+- name: Display the current authenticated user
+    oc_cmd:
+    command: whoami -c
+- name: Display the current authenticated user and get cluster information
+      oc:
+        cmd: |
+          whoami -c
+          cluster-info
+'''
+
+RETURN = r'''
+success:
+    description: Status of the module
+    type: bool
+    returned: always
+    sample: True
+stdout:
+    description: List with the command execution result/s
+    type: list
+    returned: always
+'''
+
+
+class ActionModule(ActionBase):
+
+    def run(self, tmp=None, task_vars=None):
+        super().run(tmp, task_vars)
+        module_args = self._task.args.copy()
+
+        if 'cmd' not in module_args or not module_args['cmd']:
+            raise AnsibleError('"cmd" parameter is mandatory')
+
+        # Get remote home, local or remote depending of the hosts
+        user_home = Path(task_vars['ansible_user_dir']).resolve()
```

Comment:
Keep in mind that this will fail if gather_facts is false, so I'll prefer this line to be evaluated only if really needed, in case that KUBECONFIG is not set as an environment var or cifmw_kubeconfig is not present.

Reply:
I changed the gathering to `explicit` and it still works. Any idea?